emulate.c

```diff
@@ -134,6 +134,14 @@ static void rv_except_illegal_insn(struct riscv_t *rv, uint32_t insn)
     rv->csr_mcause = code;
 }
 
+/* no trap supports on floating point exceptions */
+/* TODO: implement fp_exception functions */
+static void rv_except_fp_invalid_operation(struct riscv_t *rv);
```

Comment:
Don't put incomplete functions here. If you want to declare something, ensure the availability.

Reply:
Also, make git commit messages informative!